ssr/src/utils/notifications/mod.rs

```diff
@@ -0,0 +1,28 @@
+use device_id::send_principal_and_token_offchain;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::JsFuture;
+
+pub mod device_id;
+
+#[wasm_bindgen(module = "/src/utils/notifications/setup-firebase-messaging.js")]
+extern "C" {
+    fn get_token(vapidKey: String) -> js_sys::Promise;
+}
+
+pub async fn get_token_for_principal(principal_id: String) {
+    let token_promise = get_token(env!("vapidKey").to_string());
```

Comment:
Since the VAPID key is safe to keep public, lets put it in the repo?

Reply:
not a fan of having it in rust code, so i put it inside the messaging setup js file, along with other variables and ids